hashmap.js

```diff
@@ -25,7 +25,14 @@
 		this.clear();
 		switch (arguments.length) {
 			case 0: break;
-			case 1: this.copy(other); break;
+			case 1: {
+				if(Array.isArray(other)) {
+					this.from2DArray(other);

```

Comment:
I think the multi function won't work for this (at least not without major modifications), because calling it like `multi([ ['k1', 'v1'], ['k2', 'v2'] ])` is ambiguous: Either it creates the equivalent of `{'k1': 'v1', 'k2': 'v2'}` (ES6 map style constructor) or `{['k1', 'v1']: ['k2':'v2']}` (compat style constructor).

So either you have to pass an additional argument to multi (i.e. whether to use the compat style or the ES6 map style input) or you have separate methods (i.e. `from2DArray`).

I think adding an argument to `multi` is quite messy. However, I think `from2DArray` should probably be exposed the same way as `multi` (as a function in the utils section). What do you think?